cassandra/connection.py

```diff
@@ -764,6 +781,8 @@ def __init__(self, host='127.0.0.1', port=9042, authenticator=None,
         self._io_buffer = _ConnectionIOBuffer(self)
         self._continuous_paging_sessions = {}
         self._socket_writable = True
+        self.orphaned_request_ids = set()
```

Comment:
The drawback for using a straight set here is that req_ids are Python ints which are (usually) 4 bytes in size.  So you're looking at (in the worst case) 32768 msgs * 4 bytes/msg ~= 128k _per connection_.  In practice I don't think this will be a problem but there's no need to rely on that.  If we use a bitset here we'd normalize that to a fixed size of ~4k/conn.  Might be worth thinking about.

Reply:
Nevermind, I forgot you had a threshold for connection termination that effectively functions as a cap on this.  As long as that size doesn't become absurdly large there's no issue here.